misk/src/main/kotlin/misk/web/BoundAction.kt

```diff
@@ -114,25 +121,47 @@ private fun HttpServletRequest.accepts(): List<MediaRange> {
     }
 }
 
-/** Matches a request to an action. Can be sorted to pick the most specific match amongst a set of candidates */
-internal class RequestMatch(
-        val action: BoundAction<*, *>,
-        private val pathMatcher: Matcher,
-        private val responseContentTypeMatch: MediaRange.Matcher?
+/** Matches a request . Can be sorted to pick the most specific match amongst a set of candidates */
+internal open class RequestMatch(
+        private val pathPattern: PathPattern,
+        private val acceptedMediaRange: MediaRange,
+        private val requestCharsetMatch: Boolean,
+        private val responseContentType: MediaType
 ) : Comparable<RequestMatch> {
+
     override fun compareTo(other: RequestMatch): Int {
-        val patternDiff = action.pathPattern.compareTo(other.action.pathPattern)
+        val patternDiff = pathPattern.compareTo(other.pathPattern)
```

Comment:
What about using some kind of comparison chain? 

```
compareBy<RequestMatch>({ it.pathPattern}, {it.acceptedMediaRange}, ...).compare(this, other)
```

Reply:
interesting, yeah, could do

Reply:
although tbh i sortof prefer to have request match be directly ``Comparable``

Reply:
sure, i'm just suggesting the impl of `BoundAction#compareTo` be a call to `compareBy`

Reply:
Ah, well, it gets to be tricker to combine ``compareBy`` with explicit comparison functions which is what's needed for the media type comparison and the charset comparisons (you need to declare anonymous types etc). I think having it as straight line code is easier to read